File: Benchmarks/IONumber/Java/IONumber.java

import java.io.FileNotFoundException;
import java.io.IOException;
import java.nio.file.Files;
import java.nio.file.NoSuchFileException;
import java.nio.file.Paths;

class IONumber {
    public static int ReadValue(String path) {
        try {
            String contents = Files.readString(Paths.get(path));
            return Integer.parseInt(contents);
        } catch (IOException e) {
            return 0;
        }
    }

    public static void WriteValue(String path, int value) throws IOException {
        Files.write(Paths.get(path), String.valueOf(value + 1).getBytes());
    }

    public static void Run(int count, String path) throws IOException {
        try {
            Files.delete(Paths.get(path));
        } catch (NoSuchFileException e) {
            // Skip
        }

        int value = 0;
        while (value != count) {
            value = ReadValue(path);
            WriteValue(path, value);
        }
    }

    public static void main(String[] args) throws FileNotFoundException, IOException {
        int count = Integer.parseInt(args[0]);
        String path = args[1];

        IONumber.Run(count, path);
    }
}

File: Benchmarks/IONumber/Python/ionumber.py
import os
import sys


def read_value(path: str) -> int:
    try:
        with open(path, "r") as file:
            return int(file.read())
    except:
        return 0


def write_value(path: str, value: int) -> None:
    with open(path, "w") as file:
        file.write(str(value))


def run(count: int, path: str) -> None:
    try:
        os.remove(path)
    except:
        ...

    value = 0
    while value != count:
        value = read_value(path)
        write_value(path, value + 1)


if __name__ == "__main__":
    argv = sys.argv
    count = int(argv[1])
    path = argv[2]

    run(count, path)
